components/dashboard/leeway.Dockerfile

```diff
@@ -2,13 +2,27 @@
 # Licensed under the GNU Affero General Public License (AGPL).
 # See License-AGPL.txt in the project root for license information.
 
+FROM alpine:3.13 as compress
 
-FROM caddy/caddy:2.4.0-beta.2-alpine
+RUN apk add brotli gzip
 
-COPY components-dashboard--static/conf/Caddyfile /etc/caddy/Caddyfile
 COPY components-dashboard--app/build /www
 
+WORKDIR /www
+
+RUN find . -type f \( -name '*.html' -o -name '*.js' -o -name '*.css' -o -name '*.png' -o -name '*.svg' -o -name '*.map' -o -name '*.json' \) \
+  -exec /bin/sh -c 'gzip -v -f -9 -c "$1" > "$1.gz"' /bin/sh {} \;
+
+RUN find . -type f \( -name '*.html' -o -name '*.js' -o -name '*.css' -o -name '*.png' -o -name '*.svg' -o -name '*.map' -o -name '*.json' \) \
+  -exec /bin/sh -c 'brotli -q 11 -o "$1.br" "$1"' /bin/sh {} \;
+
+FROM caddy/caddy:2.4.0-alpine
+
+COPY components-dashboard--static/conf/Caddyfile /etc/caddy/Caddyfile
+COPY --from=compress /www /www
+
 RUN mkdir -p /www/static/bin
+
 COPY components-local-app--app/components-local-app--app-linux/local-app /www/static/bin/gitpod-local-companion-linux
 COPY components-local-app--app/components-local-app--app-darwin/local-app /www/static/bin/gitpod-local-companion-darwin
 COPY components-local-app--app/components-local-app--app-windows/local-app.exe /www/static/bin/gitpod-local-companion-windows.exe
```

Comment:
Considering those are nicely compressible Go binaries, does it might make sense to pre-compress them, too?
We could certainly pre-arrange the `/www` directory in the `compress` stage, making the dashboard image a bit neater.

Reply:
done 

```
-rwxr-xr-x    1 root     root      16352976 May 21 02:46 gitpod-local-companion-darwin
-rwxr-xr-x    1 root     root       7843185 May 21 02:46 gitpod-local-companion-darwin.gz
-rwxr-xr-x    1 root     root      17023488 May 21 02:46 gitpod-local-companion-windows.exe
-rwxr-xr-x    1 root     root       8128762 May 21 02:46 gitpod-local-companion-windows.exe.gz
-rwxr-xr-x    1 root     root      17108979 May 21 02:46 gitpod-local-companion-linux
-rwxr-xr-x    1 root     root       8237941 May 21 02:46 gitpod-local-companion-linux.gz
```